src/main/java/org/jenkinsci/plugins/workflow/actions/FlowNodeStatusAction.java

```diff
@@ -0,0 +1,34 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.Result;
+
+import javax.annotation.Nonnull;
+
+public final class FlowNodeStatusAction implements PersistentAction {
+    private final Result result;
```

Comment:
fwiw, we might want to actually go with moving [`GenericStatus`](https://github.com/jenkinsci/pipeline-graph-analysis-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/GenericStatus.java) into `workflow-api` and using that instead of `Result`.

Reply:
I kind of like that idea actually.

Reply:
Yeah, that might make some sense. Not sure what @jglick would think though, maybe there's a concrete reason not to -- we might need to provide a mapping back and forth from Result though.

Reply:
Oh, we'll definitely need that mapping. I'm assuming that if we move `GenericStatus` here, we'll also be adding logic to do that mapping.